File: frontend/src/types/missao-bateria.ts
/** TypeScript interfaces for MissaoBateria entity matching Pydantic schemas. */

export interface MissaoBateriaResponse {
  id: number;
  missao_id: number;
  bateria_id: number;
  ordem_uso: number;
  created_at: string;
}

export interface MissaoBateriaCreate {
  missao_id: number;
  bateria_id: number;
  ordem_uso: number;
}

File: app/schemas/missao_bateria.py
"""Pydantic schemas for MissaoBateria entity."""

from datetime import datetime

from pydantic import BaseModel, Field


class MissaoBateriaBase(BaseModel):
    missao_id: int
    bateria_id: int
    ordem_uso: int = Field(..., gt=0)


class MissaoBateriaCreate(MissaoBateriaBase):
    pass


class MissaoBateriaResponse(MissaoBateriaBase):
    id: int
    created_at: datetime

    model_config = {"from_attributes": True}
